### PR TITLE
Test memleak fixes

### PR DESCRIPTION
**Pull request by akallabeth**
Fixes for #4238

### CODE REVIEW

File: winpr/libwinpr/sspi/sspi_winpr.c [hunk 1]
```diff
@@ -182,18 +182,22 @@ int sspi_ContextBufferAllocTableGrow()
 	return 1;
 }
 
-void sspi_ContextBufferAllocTableFree()
+static void sspi_ContextBufferAllocTableFree(void)
 {
-	ContextBufferAllocTable.cEntries = ContextBufferAllocTable.cMaxEntries = 0;
-	free(ContextBufferAllocTable.entries);
+	if (ContextBufferAllocTable.cEntries == 0)
```
Comment (mfleisz) [inline winpr/libwinpr/sspi/sspi_winpr.c]: This looks wrong ... why would we want to free the table only if there are no entries in it?

File: winpr/libwinpr/nt/test/TestNtCreateFile.c [hunk 2]
```diff
@@ -60,23 +54,23 @@ int TestNtCreateFile(int argc, char* argv[])
 	}
 
 	result = 0;
-
 out:
-
+	_RtlFreeUnicodeString(&uString);
```
Comment (mfleisz) [inline winpr/libwinpr/nt/test/TestNtCreateFile.c]: uString is never initialized if _RtlAnsiStringToUnicodeString fails and we will end up freeing an invalid pointer here

**Comment by mfleisz** [comment]
Oh - and it looks like some of the atoi changes are part of this PR as well (which I guess results in the merge conflict) ...

**Comment by mfleisz** [comment]
@akallabeth Reviewed again - comments are still valid

File: winpr/libwinpr/nt/test/TestNtCreateFile.c (hunk 2 above)
Comment (akallabeth) [reply winpr/libwinpr/nt/test/TestNtCreateFile.c]: Wrong, see line 25 (_RtlAnsiStringToUnicodeString)

**Comment by akallabeth** [comment]
@hardening @bmiklautz Anyone got time for this one?

File: libfreerdp/codec/test/TestFreeRDPCodecZGfx.c [hunk 3]
```diff
@@ -29,10 +29,10 @@ static int test_ZGfxCompressFox(void)
 	int rc = -1;
 	int status;
 	UINT32 Flags;
-	BYTE* pSrcData;
+	BYTE* pSrcData = NULL;
 	UINT32 SrcSize;
```
Comment (hardening) [inline libfreerdp/codec/test/TestFreeRDPCodecZGfx.c]: This initialization isn't needed.

**Pull request merged**